File: src/main/java/arraysandstrings/LengthOfLongestSubarray.java
package arraysandstrings;

/**
 * Given an array of positive integers nums and an integer k,
 * find the length of the longest subarray whose sum is less than or equal to k.
 */

public class LengthOfLongestSubarray {
    public static int findLength(int [] nums, int k) {
        int left = 0;
        int curr = 0;
        int ans = 0;

        for (int right = 0; right < nums.length - 1; right++) {
            curr += nums[right];

            while (curr > k) {
                curr -= nums[left];
                left++;
            }

            ans = Math.max(ans, right - left - 1);
        }

        return ans;
    }
}

File: src/kotlin/arraysandstrings/LengthOfLongestSubarray.kt
package arraysandstrings

import kotlin.math.max

fun findLength(a: Array<Int>, k: Int): Int {
    var left = 0
    var c = 0
    var r = 0

    for (right in 0 .. a.size - 1) {
        c += a[right]

        while (c > k) {
            c -= a[left]
            left++
        }

        r = max(r, right - left - 1)
    }

    return r
}